HACKING.md

```diff
@@ -280,14 +280,13 @@ Once the patch set is ready to be reviewed, push the branch to the public remote
 
 ### Development environment and source code navigation
 
-Scylla includes a [CMake](https://cmake.org/) file, `CMakeLists.txt`, for use only with development environments (not for building) so that they can properly analyze the source code.
+Scylla uses [CMake](https://cmake.org/) and ninja as its building system. This allows for seamless
+integration with any IDE or tool that supports CMake projects or can utilize `compile_commands.json`.
```

Comment:
Hmmm, distcc may already be documented sufficiently: said part of this document says to set
```
export CCACHE_PREFIX="distcc"
```
and that way ccache takes care of distcc.

Based on that, it might suffice just to recommend

```
export CMAKE_CXX_COMPILER_LAUNCHER=/usr/bin/ccache
export CMAKE_C_COMPILER_LAUNCHER=/usr/bin/ccache
```

Reply:
- do we need to extend install-dependencies.sh to install cmake?
- do we need to update these documents (wherever they speak about dependencies / install-dependencies) to also mention cmake?

Reply:
- going forward, are we going to include cmake in the frozen toolchain (dbuild) container images? (well this question applies more to the README, not HACKING, sorry)

Reply:
> * do we need to extend install-dependencies.sh to install cmake?

it's already extended. see https://github.com/scylladb/scylladb/blob/298cafff3537950f1b17b1fcf119e3d831bcd786/install-dependencies.sh#L304

and

- https://github.com/scylladb/seastar/blob/47c52cb3e19c4757093b812876ea245d7fc0b62d/install-dependencies.sh#L31
- https://github.com/scylladb/seastar/blob/47c52cb3e19c4757093b812876ea245d7fc0b62d/install-dependencies.sh#L82
- https://github.com/scylladb/seastar/blob/47c52cb3e19c4757093b812876ea245d7fc0b62d/install-dependencies.sh#L170

> 
>     * do we need to update these documents (wherever they speak about dependencies / install-dependencies) to also mention cmake?

now that it's already covered by seastar's `install-dependencies.sh`. i think CMake is part of the building dependencies, unless we are picky on cmake versions or configurations. i don't think we should single it out, and talk about it specifically.